docs/current/api/975146-concepts.md

```diff
@@ -0,0 +1,126 @@
+      exec(args: ["apk", "info"]) {
+        stdout {
+          contents
+        }
+      }
+    }
+  }
+}
+```
+
+This query represents a very simple Dagger pipeline. In plain English, it instructs Dagger to "download the latest `alpine` container image, run the command `apk info` in that image, and print the results of the command to the standard output device".
+
+Here's an example of the command output: a list of the packages installed in the image:
+
+```graphql
+{
+ "data": {
+  "container": {
+   "from": {
+    "exec": {
+     "stdout": {
+      "contents": "alpine-baselayout-data\nmusl\nbusybox\nalpine-baselayout\nalpine-keys\nca-certificates-bundle\nlibcrypto1.1\nlibssl1.1\nssl_client\nzlib\napk-tools\nscanelf\nmusl-utils\nlibc-utils\n"
+     }
+    }
+   }
+  }
+ }
+}
+```
+
+A GraphQL schema works by defining one or more object types, and then fields (and field arguments) on those object types. The fields of an object type can themselves be objects, allowing for different entities to be logically connected with each other. API users can then perform GraphQL queries to return all or some fields of an object.
+
+With Dagger's GraphQL API, while all of the above is true, each field in a query also resolves to a build operation. To understand this, let's dissect the query above:
+
+1. Consider the first level of the previous query:
+
+  ```graphql
+  query {
+    container {
+      from(address: "alpine:latest") {
+        ...
+      }
+    }
+  }
+  ```
+
+  This query requests the `from` field of Dagger's `Container` object type, passing it the address of a container image as argument. To resolve this, Dagger will initialize a container from the image published at the given address and return a `Container` object representing the new container image.
+
+1. Now, consider the next level of the query:
+
+  ```graphql
+  query {
+    container {
+      from(address: "alpine:latest") {
+        exec(args: ["apk", "info"]) {
+          ...
+        }
+      }
+    }
+  }
+  ```
+
+  Here, the query requests the `exec` field of the `Container` object returned in the previous step, passing it the command to be executed as an array of arguments. To resolve this, Dagger will  define the command for execution in the container image and return a `Container` object containing the execution results.
+
+1. Finally, consider the innermost level of the query:
+
+  ```graphql
+  query {
+    container {
+      from(address: "alpine:latest") {
+        exec(args: ["apk", "info"]) {
+          stdout {
+            contents
+          }
+        }
+      }
+    }
+  }
+  ```
+
+  Here, the query requests the `stdout` field of the `Container` object returned in the previous step. To resolve this, Dagger will return a `File` object type containing the result of the last executed command. The output of the command is retrieved by addressing the `contents` field of the `File` type, which returns a `String`.
+
+## Lazy-Loading
```

Comment:
I would like feedback on which concepts should be discussed in this guide. The ones I've planned are lazy-loading and chaining, but maybe there are other (better?) ones?